### PR TITLE
Added ability to pass AWS profiles as arguments.

### PR DESCRIPTION
**Pull request by Caid11**
Added a new parameter, `profile_name`, to the following methods:
- `getHighestVersion`
- `listSecrets`
- `putSecret`
- `getAllSecrets`
- `getSecret`
- `deleteSecrets`
- `createDdbTable`

This parameter allows the user to pass in a profile name as an optional argument. This profile name is then used to generate an AWS Boto session, which is then used to create clients and resources using the credentials in the given profile.

Basically, it lets the user specify the AWS profile they'd like to use when calling the listed methods.

### CODE REVIEW

**Comment by Caid11** [comment]
Whoops, found an error. I guess the dynamodb resource doesn't behave properly when generated from a boto3 session constructed with a profile.

**Comment by mannickutd** [comment]
Code looks fine. No tests but probably a bit overkill.

**Comment by alex-luminal** [comment]
Looks good.
